File: LeetCode/src/1186.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maximumSum = function(arr) {
    if (arr.length == 0) return 0;

    const noDeletion = Array(arr.length);
    noDeletion[0] = arr[0];
    for (let i = 1; i < arr.length; i++) {
        noDeletion[i] = Math.max(arr[i], arr[i] + noDeletion[i-1])
    }
    
    const oneDeletion = Array(arr.length);
    oneDeletion[0] = arr[0];
    for (let i = 1; i < arr.length; i++) {
        oneDeletion[i] = Math.max(noDeletion[i], noDeletion[i-1], arr[i] + oneDeletion[i-1])
    }
    
    
    return Math.max.apply(null, oneDeletion);
};

File: LeetCode/src/1186.py
class Solution:
    def maximumSum(self, arr: List[int]) -> int:
        noDeletion = [arr[0]]
        for i in range(1, len(arr)):
            noDeletion.append(max(arr[i], arr[i] + noDeletion[i-1]))
        
        oneDeletion = [arr[0]]
        for i in range(1, len(arr)):
            oneDeletion.append(max(noDeletion[i], noDeletion[i-1], arr[i] + oneDeletion[i-1]))
        
        return max(oneDeletion)
